x-pack/plugins/security_solution/server/endpoint/services/workflow_insights/helpers.ts

```diff
@@ -60,3 +65,34 @@ export async function createPipeline(esClient: ElasticsearchClient): Promise<boo
   });
   return response.acknowledged;
 }
+
+const validKeys = new Set([
+  'ids',
+  'categories',
+  'types',
+  'sourceTypes',
+  'sourceIds',
+  'targetTypes',
+  'targetIds',
+  'actionTypes',
+]);
+const paramFieldMap = {
+  ids: '_id',
+  sourceTypes: 'source.type',
+  sourceIds: 'source.id',
+  targetTypes: 'target.type',
+  targetIds: 'target.id',
+  actionTypes: 'action.type',
+};
+export function buildEsQueryParams(searchParams: SearchParams): QueryDslQueryContainer[] {
+  return Object.entries(searchParams).reduce((acc: object[], [k, v]) => {
+    if (!validKeys.has(k)) {
+      return acc;
+    }
+
+    const paramKey = _get(paramFieldMap, k, k);
+    const next = { terms: { [paramKey]: v } };
+
+    return [...acc, next];
```

Comment:
nit: iirc there was an effort to change all object/array creations to mutating the accumulator  in all `reduce()` functions, in order to reduce overall footprint. i think it would be better to switch to that here as well